conan/cli/commands/lock.py

```diff
@@ -179,3 +183,101 @@ def lock_update(conan_api, parser, subparser, *args):
     lockfile.update(requires=args.requires, build_requires=args.build_requires,
                     python_requires=args.python_requires, config_requires=args.config_requires)
     conan_api.lockfile.save_lockfile(lockfile, args.lockfile_out)
+
+
+
+
+@conan_subcommand()
+def lock_upgrade(conan_api, parser, subparser, *args):
+    """
+    (Experimental) Upgrade requires, build-requires or python-requires from an existing lockfile given a conanfile
+    or a reference.
+    """
+
+    """
+    TODOs:
+    - [ ] args.update should always be True?
+    - [x] manage input version ranges
+    - [ ] Consider a new option to enable transitive dependencies on different kinds
+          (eg. --transitive=all, --transitive=python-requires, ...)
+    - [ ] Improve the detection of range versions
+    """
+
+    common_graph_args(subparser)
+    subparser.add_argument('--update-requires', action="append", help='Update requires from lockfile')
+    subparser.add_argument('--update-build-requires', action="append", help='Update build-requires from lockfile')
+    subparser.add_argument('--update-python-requires', action="append", help='Update python-requires from lockfile')
+    subparser.add_argument('--update-config-requires', action="append", help='Update config-requires from lockfile')
+    subparser.add_argument('--build-require', action='store_true', default=False, help='Whether the provided reference is a build-require')
+    subparser.add_argument('--transitives', action='store_true', default=False, help='Upgrade also transitive dependencies')
```

Comment:
I'd probably drop the ``--transitives`` argument at the moment and focus on just 1 use case to start with, start more minimalistic.

Reply:
I've just removed in the latest commit to simplify things for now